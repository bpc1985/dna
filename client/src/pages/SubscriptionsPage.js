import React, { useState, useEffect } from "react";
import Search from "../components/Search/Search";
import ListCard from "../components/Cards/ListCard";
import Loading from '../components/Loading/Loading'
import { fetchData, isSearchTermIncluded } from "../utils";

import "./SubscriptionsPage.scss";

const initalState = {
  searchTerm: "",
  subscriptions: null,
  filteredSubscriptions: null
};

export default function SubscriptionsPage(props) {
  const [state, setState] = useState(initalState);

  useEffect(() => {
    async function fetchSubscriptionsData() {
      const subscriptions = await fetchData("subscriptions?_expand=dnaPackage");
      setState({
        ...state,
        subscriptions,
        filteredSubscriptions: subscriptions
      });
    }
    fetchSubscriptionsData();

    return () => {};
  }, []);

  useEffect(() => {
    if (state.subscriptions) {
      const filteredSubscriptions = state.subscriptions.filter(sub =>
        isSearchTermIncluded(sub.dnaPackage, state.searchTerm)
      );
      setState({ ...state, filteredSubscriptions });
    }
  }, [state.searchTerm]);

  const onDetail = sid => {
    props.history.push(`/subscriptions/${sid}`);
  };

  const onSearch = searchTerm => {
    setState({ ...state, searchTerm });
  };

  const { searchTerm, filteredSubscriptions } = state;

  if (!filteredSubscriptions) {
    return <Loading />;
  }

  return (
    <React.Fragment>
      <div className="searchGroup">
        <div className="pageTitle">Subscriptions</div>
        <Search searchTerm={searchTerm} onSearch={onSearch} />
      </div>
      <ListCard data={filteredSubscriptions} onDetail={onDetail} />
    </React.Fragment>
  );
}
