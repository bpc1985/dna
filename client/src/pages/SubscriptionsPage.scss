.searchGroup {
  display: flex;
  justify-content: space-between;

  .pageTitle {
    font-size: 1.5rem;
    font-weight: bold;
    margin: 5px 0;
  }
}

@media (max-width: 420px) {
  .searchGroup {
    display: block;
  }
}