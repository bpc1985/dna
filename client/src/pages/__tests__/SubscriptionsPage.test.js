import { render, fireEvent, wait } from "@testing-library/react";
import React from "react";
import SubscriptionsPage from "../SubscriptionsPage";

const props = {
  history: {
    push: jest.fn()
  }
};

jest.mock("../../utils", () => {
  const data = [
    {
      _id: "111",
      phone_number: "111111111",
      address: "Address 1",
      dnaPackage: {
        name: "Package 1",
        description: "Description 1",
        type: "mobile"
      }
    },
    {
      _id: "222",
      phone_number: "2222222222",
      address: "Address 2",
      dnaPackage: {
        name: "Package 2",
        description: "Description 2",
        type: "boardband"
      }
    }
  ];

  return {
    fetchSubscriptionsData: () => Promise.resolve(data),
    isSearchTermIncluded: () => true
  };
});

describe("SubscriptionsPage", () => {
  it("should render component", async () => {
    const wrapper = render(<SubscriptionsPage history={props.history} />);
    await wait(() => {
      expect(wrapper.container).toMatchSnapshot();
    });
  });
});
