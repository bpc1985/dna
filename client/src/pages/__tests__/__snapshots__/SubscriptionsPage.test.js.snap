// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubscriptionsPage should render component 1`] = `
<div>
  <div
    class="searchGroup"
  >
    <div
      class="pageTitle"
    >
      Subscriptions
    </div>
    <div>
      <input
        class="searchInput"
        name="searchInput"
        placeholder="Search your package"
        type="text"
        value=""
      />
    </div>
  </div>
  <ul
    class="subscriptionList"
  >
    <li
      class="subscriptionItem"
    >
      <div>
        <h1>
          Package 1
        </h1>
        <h2>
          111111111
        </h2>
      </div>
      <div>
        <span>
          mobile
        </span>
        <button
          class="btn"
        >
          Details
        </button>
      </div>
    </li>
    <li
      class="subscriptionItem"
    >
      <div>
        <h1>
          Package 2
        </h1>
        <h2>
          Address 2
        </h2>
      </div>
      <div>
        <span>
          boardband
        </span>
        <button
          class="btn"
        >
          Details
        </button>
      </div>
    </li>
  </ul>
</div>
`;
